### PR TITLE
ui-manchette: highlight selected train

### LINKED ISSUE #268: ui-manchette: projectedTrain can have duplicate keys
To fix this problem, the usePaths hook must be modified to use train's id for the path.
The type ProjectPathTrainResult must be modified to support id.

### LINKED ISSUE #292: Manchette-Get: pouvoir visualiser le train sélectionné dans le GET
Acceptance criteria

- [x] Le train dont l'id est sélectionné se met en surbrillance dans le GET

### PR DESCRIPTION
**Pull request by Math-R**
- Now the component also use the real name of trains 

close #268
close https://github.com/OpenRailAssociation/osrd-ui/issues/292

### CODE REVIEW

**Review by clarani — APPROVED** [review]
LGTM, tested :white_check_mark:

File: ui-manchette/src/components/Manchette.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ import { useIsOverflow } from '../hooks/useIsOverFlow';
 type ManchetteProps = {
   operationalPoints: OperationalPointType[];
   projectPathTrainResult: ProjectPathTrainResult[];
+  selectedProjection: number;
```
Comment (clarani) [inline ui-manchette/src/components/Manchette.tsx]: The selectedProjection should be nullable

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Review by Yohh — APPROVED** [review]
LGTM 💪

**Pull request merged**